### PR TITLE
refactor(env): simplify ternary statement for dsync feature flag

### PR DESCRIPTION
**Pull request by Musyonchez**
This PR addresses a code style issues in the `env.ts` configuration file. Previously, you were using an unneeded ternary statement to set the flag. However, this can be simplified to a single logical NOT operation.

To improve readability, we've updated the code to avoid the unneeded ternary statement.

Changes include:
- Updating the feature flag in `env.ts` to use a logical NOT operation instead of a ternary statement.

This change should make the code in `env.ts` easier to read and maintain.

### CODE REVIEW

File: lib/env.ts [hunk 1]
```diff
@@ -78,31 +78,29 @@ const env = {
   },
 
   disableNonBusinessEmailSignup:
-    process.env.DISABLE_NON_BUSINESS_EMAIL_SIGNUP === 'true' ? true : false,
+    !!(process.env.DISABLE_NON_BUSINESS_EMAIL_SIGNUP === 'true'),
```
Comment (ukrocks007) [inline lib/env.ts:81]:
```suggestion
    process.env.DISABLE_NON_BUSINESS_EMAIL_SIGNUP === 'true',
```

File: lib/env.ts [hunk 2]
```diff
@@ -78,31 +78,29 @@ const env = {
...
 
   authProviders: process.env.AUTH_PROVIDERS || 'github,credentials',
 
   otel: {
     prefix: process.env.OTEL_PREFIX || 'boxyhq.saas',
   },
 
-  hideLandingPage: process.env.HIDE_LANDING_PAGE === 'true' ? true : false,
+  hideLandingPage: !!(process.env.HIDE_LANDING_PAGE === 'true'),
```
Comment (ukrocks007) [inline lib/env.ts:89]:
```suggestion
  hideLandingPage: process.env.HIDE_LANDING_PAGE === 'true',
```

File: lib/env.ts [hunk 3]
```diff
@@ -78,31 +78,29 @@ const env = {
...
-  darkModeEnabled: process.env.NEXT_PUBLIC_DARK_MODE === 'false' ? false : true,
+  darkModeEnabled: !(process.env.NEXT_PUBLIC_DARK_MODE === 'false'),
 
   teamFeatures: {
-    sso: process.env.FEATURE_TEAM_SSO === 'false' ? false : true,
-    dsync: process.env.FEATURE_TEAM_DSYNC === 'false' ? false : true,
-    webhook: process.env.FEATURE_TEAM_WEBHOOK === 'false' ? false : true,
-    apiKey: process.env.FEATURE_TEAM_API_KEY === 'false' ? false : true,
-    auditLog: process.env.FEATURE_TEAM_AUDIT_LOG === 'false' ? false : true,
+    sso: !(process.env.FEATURE_TEAM_SSO === 'false'),
```
Comment (ukrocks007) [inline lib/env.ts:94]:
```suggestion
    sso: process.env.FEATURE_TEAM_SSO !== 'false',
```

File: lib/env.ts [hunk 4]
```diff
@@ -78,31 +78,29 @@ const env = {
...
+    apiKey: !(process.env.FEATURE_TEAM_API_KEY === 'false'),
+    auditLog: !(process.env.FEATURE_TEAM_AUDIT_LOG === 'false'),
     payments:
       process.env.FEATURE_TEAM_PAYMENTS === 'false'
         ? false
-        : process.env.STRIPE_SECRET_KEY && process.env.STRIPE_WEBHOOK_SECRET
-          ? true
-          : false,
-    deleteTeam: process.env.FEATURE_TEAM_DELETION === 'false' ? false : true,
+        : !!(process.env.STRIPE_SECRET_KEY && process.env.STRIPE_WEBHOOK_SECRET),
```
Comment (ukrocks007) [inline lib/env.ts:102]:
```suggestion
        : Boolean(process.env.STRIPE_SECRET_KEY && process.env.STRIPE_WEBHOOK_SECRET),
```

**Review by ukrocks007 — CHANGES_REQUESTED** [review]

**Comment by Musyonchez** [comment]
I am in a place with slow internet i will fix the errors tomorrow.

**Comment by Musyonchez** [comment]
I have made the changes as suggested by @ukrocks007. Please check.

**Review by ukrocks007 — APPROVED** [review]

**Review by deepakprabhakara — APPROVED** [review]

**Pull request merged**